GrandCentralBoard/Widgets/Watch/Server/GoogleCalendarDataProvider.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  GoogleCalendarDataProvider.swift
+//  GrandCentralBoard
+//
+//  Created by Michał Laskowski on 07.04.2016.
+//  Copyright © 2016 Oktawian Chojnacki. All rights reserved.
+//
+
+import Foundation
+import Result
+
+protocol CalendarDataProviding {
+    func fetchEventsForCalendar(calendarID: String, completion: (Result<[EventModel], APIDataError>) -> Void)
+    func fetchCalendar(calendarID: String, completion: (Result<CalendarModel, APIDataError>) -> Void)
+}
+
+private enum CalendarAPIAction : String {
+    case GetDetails = ""
+    case GetEvents  = "events"
+
+    static private let baseURL = NSURL(string: "https://www.googleapis.com/calendar/v3/calendars/")!
+
+    func URLForCalendar(calendarID: String) -> NSURL? {
+        guard let escapedCalendarID = calendarID.URLEscape() else {
+            assertionFailure("Failure escaping calendar id: \(calendarID)")
+            return nil
+        }
+        return self.dynamicType.baseURL.URLByAppendingPathComponent("\(escapedCalendarID)/\(self.rawValue)")
```

Comment:
It's a static property, I could only replace `self.dynamicType` with `CalendarAPIAction`. I will leave it as it is